installer.sh.in

```diff
@@ -801,22 +818,30 @@ set_bootloader() {
     chroot $TARGETDIR grub-install $grub_args $dev >$LOG 2>&1
     if [ $? -ne 0 ]; then
         DIALOG --msgbox "${BOLD}${RED}ERROR:${RESET} \
-        failed to install GRUB to $dev!\nCheck $LOG for errors." ${MSGBOXSIZE}
+failed to install GRUB to $dev!\nCheck $LOG for errors." ${MSGBOXSIZE}
         DIE 1
     fi
     echo "Running grub-mkconfig on $TARGETDIR..." >$LOG
     chroot $TARGETDIR grub-mkconfig -o /boot/grub/grub.cfg >$LOG 2>&1
     if [ $? -ne 0 ]; then
         DIALOG --msgbox "${BOLD}${RED}ERROR${RESET}: \
-        failed to run grub-mkconfig!\nCheck $LOG for errors." ${MSGBOXSIZE}
+failed to run grub-mkconfig!\nCheck $LOG for errors." ${MSGBOXSIZE}
         DIE 1
     fi
 }
 
 test_network() {
-    rm -f xtraeme.asc && \
-        xbps-uhelper fetch http://alpha.de.repo.voidlinux.org/live/xtraeme.asc >$LOG 2>&1
-    if [ $? -eq 0 ]; then
+    # Pick a peace of data to download to ensure that a network connection is
+    # available.
+    local file="http://alpha.de.repo.voidlinux.org/live/xtraeme.asc"
+
+    # Download file with XBPS, remove it directly afterwards
+    xbps-uhelper fetch "$file" >$LOG 2>&1
+    local status=$?
+    rm -f xtraeme.asc
```

Comment:
I guess this works too?

Reply:
I've tested this solution in VMs and on bare metal on my notebook and desktop PC and it worked as expected. If you have no internet connection the download fails and the proper message is shown to the user, if the download succeeded the success message appears and the currently downloaded xtraeme.asc in the home directory will be deleted right after the check, so that the home directory stay cleaned up.